artifacts/deploy/karmada-secret-config.yaml

```
@@ -0,0 +1,26 @@
+apiVersion: v1
+stringData:
+  karmada.config: |-
+    apiVersion: v1
+    clusters:
+    - cluster:
+        certificate-authority-data: ${ca_crt}
+        server: https://karmada-apiserver.karmada-system.svc.cluster.local:5443
+      name: karmada-apiserver
+    contexts:
+    - context:
+        cluster: karmada-apiserver
+        user: karmada-apiserver
+      name: karmada-apiserver
+    current-context: karmada-apiserver
+    kind: Config
+    preferences: {}
+    users:
+    - name: karmada-apiserver
+      user:
+        client-certificate-data: ${client_crt}
+        client-key-data: ${client_key}
+kind: Secret
+metadata:
+  name: ${component}-karmada-config
+  namespace: karmada-system
```

```suggestion
apiVersion: v1
kind: Secret
metadata:
  name: ${component}-karmada-config
  namespace: karmada-system
stringData:
  karmada.config: |-
    apiVersion: v1
    clusters:
    - cluster:
        certificate-authority-data: ${ca_crt}
        server: https://karmada-apiserver.karmada-system.svc.cluster.local:5443
      name: karmada-apiserver
    contexts:
    - context:
        cluster: karmada-apiserver
        user: karmada-apiserver
      name: karmada-apiserver
    current-context: karmada-apiserver
    kind: Config
    preferences: {}
    users:
    - name: karmada-apiserver
      user:
        client-certificate-data: ${client_crt}
        client-key-data: ${client_key}
```
apiVersion --> kind --> metadata --> stringData